workspaces/adventure-pack/goodies/typescript/Iterator.prototype.flatMap/index.test.ts

```diff
@@ -0,0 +1,184 @@
+import { describe, expect, it } from "@jest/globals";
+
+import { iteratorPrototype } from "../Iterator.prototype";
+delete (iteratorPrototype as unknown as Record<string, unknown>).flatMap;
+// eslint-disable-next-line import-x/first -- This has to happen after we delete the built-in implementation.
+import "./index";
+
+describe("Iterator.prototype.flatMap", () => {
+  it("can flatten an Array's values()", () => {
+    const array = [-1, -5, 5, 3, 7];
+    const generator = function* (p: number): Generator<number, void, void> {
```

Comment:
Technically, this is a callback, whose return value is a generator object!

```suggestion
    const callback = function* (p: number): Generator<number, void, void> {
```

Reply:
ack!